### DISCUSSION TITLE
List changed packages in monorepo?

### DISCUSSION THREAD

**Discussion by itisnas** [pnpm/pnpm#4327] [Q&A] [answered]
Hi champs!

What is the pnpm equivalent of `lerna updated -a`?

The above script will list all the packages in your workspace that have changed since the last commit.

By packages, I'm referring to in the context of monorepo.

The closest I've been to is:

```
$  pnpm list "@desisyst/*" --filter "...[origin/master]"
```

I expect the output to be something like:

```
pnpm info Looking for changed packages since @desisyst/core@1.0.0

@desisyst/client
@desisyst/component
@desisyst/developer

pnpm success found 3 packages ready to publish
```

Thanks 🙏

**Comment by zkochan** [comment, marked as answer]
try

```
pnpm list "@desisyst/*" --filter "...[origin/master]" --depth=-1 --parseable
```

**Reply by BlackHole1** [reply]
The result is `path`, not `package name`

<img width="577" alt="image" src="https://user-images.githubusercontent.com/8198408/153792216-1aae3a7c-2ce2-434c-b72f-4175077952ec.png">

**Reply by itisnas** [reply]
Thank you! Much appreciated it 🙏

**Comment by BlackHole1** [comment]
**use `awk`**

```shell
pnpm list "@desisyst/*" --filter "...[origin/main]" --depth=-1 |  awk 'NF{print $1}' | awk -F '@' '{print $1$2}'
```

<img width="838" alt="image" src="https://user-images.githubusercontent.com/8198408/153792066-875ff374-3d81-4e31-b8a9-2124b94fc4ae.png">

----

**use `jq`(need install jq command)***

```shell
pnpm list "@desisyst/*" --filter "...[origin/main]" --depth=-1 --json | jq -r '.[] | .name'
```

<img width="678" alt="image" src="https://user-images.githubusercontent.com/8198408/153793173-c7d7a110-fdb5-4030-9fa9-17ef18bfd5f1.png">

**Comment by BlackHole1** [comment]
@zkochan Perhaps we need support: `list --only-show='name' | 'path' | 'version'` ?

**Reply by zkochan** [reply]
why is it needed? for processing? the `--json` option is also supported.
